TESTING.md

```diff
@@ -0,0 +1,69 @@
+  const { getByTestId } = render(<SomeComponent data-testid="test" text="foo" />);
+  
+  const component = await getByTestId('test');
+
+  expect(component.textContent).toEqual('foo');
+  expect(component.getAttribute('aria-label')).toEqual('foo');
+});
+```
+
+This is a "test" and not a specification, but a test that asserts multiple outputs. If any of the expectations fail, the CI will output something like "Failed: SomeComponent should render correctly". The message doesn't give any indication of why any of the expectations would fail. Further more, the failures don't give much indication as to what is failing. Without looking at the code, the failure message of the expectation will say something like "Expected 2 to equal 1" or "Expected 'bar' to equal 'foo'". Often to find out why the test fails, the source code of the test has to be consulted and the test has to be debugged. After that, there is no context to why the failure even matters. This test doesn't give much feedback or confidence on failure and thus isn't a very useful test.
+
+**Good**:
+```ts
+describe('SomeComponent', () => {
+  it('should render the "text" prop as the text', async () => {
+    const { getByTestId } = render(<SomeComponent data-testid="test" text="foo" />);
+    
+    const component = await getByTestId('test');
+    expect(component).toHaveTextContent('foo');
+  });
+  
+  it('should render the "text" prop as an aria-label for accessibility', async () => {
+    const { getByTestId } = render(<SomeComponent data-testid="test" text="foo" />);
+    
+    const component = await getByTestId('test');
+    expect(component).toHaveAttribute('aria-label', 'foo');
+  });
+});
+```
+
+This series of specifications give more context to the test. It is more code, but it is more clear what the intent of each specification is. If one fails, it will be easier to diagnose what and why. The assertions also make use of `jest-dom` which gives more context into _how_ it fails. Instead of a generic assertion, the CLI will output something like `Expected <div data-testid="test" aria-label="bar">bar</div> to have an "aria-label" of "foo" but received "bar"`. The specification should tell us which spec failed and the assertion gives insight into how.
+
+## Unit tests
+"Unit" test is not actually an accurate name since most canvas-kit unit tests include testing a DOM implementation. Canvas-kit uses [Jest](https://jestjs.io/), [React Testing Library](https://testing-library.com/docs/react-testing-library/intro) and [jest-dom](https://github.com/testing-library/jest-dom) which have a nice API for testing React components. "Unit" tests should test a React component's props. For example, if documentation says that a component will pass extra props to a container component, there should be a specification that verifies this claim. If a component's documentation states that children will be rendered, a test should verify the children exist within the containing component.
```

Comment:
```suggestion
"Unit" test is not actually an accurate name since most Canvas Kit unit tests include testing a DOM implementation. Canvas Kit uses [Jest](https://jestjs.io/), [React Testing Library](https://testing-library.com/docs/react-testing-library/intro), and [jest-dom](https://github.com/testing-library/jest-dom) which have a nice API for testing React components. "Unit" tests should test a React component's props. For example, if documentation says that a component will pass extra props to a container component, there should be a specification that verifies this claim. If a component's documentation states that children will be rendered, a test should verify the children exist within the containing component.
```